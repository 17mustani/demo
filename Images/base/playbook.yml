---
- hosts: all
  become: yes

  tasks:
  - selinux: state=disabled
  - copy: content="overlay" dest=/etc/modules-load.d/overlay.conf
  - copy:
      content: |
        net.bridge.bridge-nf-call-ip6tables = 1
        net.bridge.bridge-nf-call-iptables = 1
      dest: /usr/lib/sysctl.d/90-system.conf

  - yum: name=* state=latest
  - yum: name=NetworkManager state=latest

  - file: path=/opt/bin state=directory
  - get_url:
      url: https://github.com/kelseyhightower/setup-network-environment/releases/download/1.0.1/setup-network-environment
      dest: /opt/bin
      mode: 0755
  - copy: src=services/ dest=/etc/systemd/system/ mode=0644
  - service: name={{ item }} enabled=true
    with_items:
      - NetworkManager-wait-online
      - setup-network-environment

