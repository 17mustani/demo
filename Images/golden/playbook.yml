---
- hosts: all
  become: yes

  tasks:

  - yum_repository:
      name: Kubernetes
      description: Kubernetes Repository
      baseurl: http://yum.kubernetes.io/repos/kubernetes-el7-x86_64
      gpgcheck: no

  - yum: name={{ item }} state=latest
    with_items:
      - docker
      - kubernetes-cni
      - kubectl
      - kubelet
      - kubeadm

  - lineinfile:
      dest: /etc/sysconfig/docker-storage
      regexp: '^DOCKER_STORAGE_OPTIONS='
      line: 'DOCKER_STORAGE_OPTIONS="--storage-driver=overlay"'

  - name: Temp remove of extra args from drop-in - upstream rpm sets incorrectly
    lineinfile:
      dest: /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
      regexp: '^Environment="KUBELET_EXTRA_ARGS'
      state: absent

  - lineinfile:
      dest: /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
      insertafter: '^Environment=\"KUBELET_AUTHZ_ARGS'
      line: 'Environment="KUBELET_EXTRA_ARGS=--cgroup-driver=systemd --cloud-provider=aws"'

  - copy:
      content: |
        [Service]
        Environment="KUBELET_EXTRA_ARGS=--cloud-provider=aws"
      dest: /etc/systemd/system/kubelet.service.d/20-cloud-provider.conf

  - service: name=docker enabled=yes
  - service: name=kubelet enabled=yes

